array-reduce/main.js

```diff
@@ -0,0 +1,43 @@
+const numbers = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12];
+
+const account = [
+  { type: 'deposit', amount: 150 },
+  { type: 'deposit', amount: 20 },
+  { type: 'withdrawal', amount: 5 },
+  { type: 'deposit', amount: 100 },
+  { type: 'withdrawal', amount: 25 },
+  { type: 'withdrawal', amount: 60 }
+];
+
+const traits = [
+  { color: 'yellow' },
+  { type: 'electric' },
+  { name: 'pikachu' },
+  { level: 15 },
+  { trainer: 'ash' }
+];
+
+const sum = numbers.reduce((accumulator, currentValue) => {
+  return accumulator + currentValue;
+});
+
+const product = numbers.reduce((accumulator, currentValue) => {
+  return accumulator * currentValue;
+});
+
+const initialValue = 0;
+let withdrawal = 0;
+let deposit = 0;
```

Comment:
Instead of affecting these external variables, just add to or subtract from your `accumulator`.

Reply:
![Screen Shot 2021-03-23 at 2 34 22 PM](https://user-images.githubusercontent.com/74885918/112221850-2c2f4880-8be5-11eb-94b9-b966829e90cd.png)